src/mlib/int128.h

```diff
@@ -0,0 +1,614 @@
+#ifndef MLIB_INT128_H_INCLUDED
+#define MLIB_INT128_H_INCLUDED
+
+#include "./macros.h"
+
+#include <stdbool.h>
+#include <inttypes.h>
+#include <string.h>
+#include <stdlib.h>
+
+MLIB_C_LINKAGE_BEGIN
+
+/**
+ * @brief A 128-bit binary integer
+ */
+typedef union {
+   struct {
+      uint64_t lo;
+      uint64_t hi;
+   } r;
+#if defined(__SIZEOF_INT128__)
+   // These union members are only for the purpoes of debugging visualization
```

Comment:
typo:
```suggestion
   // These union members are only for the purpose of debugging visualization
```

(Aside, this makes the ABI of this union depend on whether `__int128_t` is available, I assume that’s okay?)

Reply:
No, but also yes, but also no: Practically no, because the union members are all 16 bytes, but yes: it may have 16byte alignment iff the __in128s are present, and technically an ODR violation if mixing TUs that differ on `__SIZEOF_INT128__`, but no: mixing TUs that would differ at that level (i.e. mixing compilers) would be a recipe for trouble, and this isn't exposed on the public API/ABI, so we're free to change it on a whim.